### ISSUE TITLE
Draft for advanced EventTypes

### ISSUE THREAD

**Issue by romainlange** [schemaorg/schemaorg#406]
Hi,

I made several contributions about EventTypes and I summed them up in an [open document](https://docs.google.com/spreadsheets/d/1CsitJZIHldzY434gi_fgeHAzxz_uhRA6AN33NYDn2C0/edit?usp=sharing).

First I would define an _event_ as a "physical gathering offered to an audience as a precise date and place".

Then I offer to organize EventTypes in _families_. For instance _performing arts_ and _visual arts_ are families, then concert, theater, painting exhibitions, are EventTypes.

As this list should remain as simple as possible, I suggest to leave some families with only one eventType.

Full disclosure: we work on Cibul.net, a platform helping to build community calendars, we are used by cities and organizers. We have a good knowledge of this issue and a standard would be useful for cross-platform communication.

Thanks for your feedback!

**Comment by mfhepp** [comment]
Thanks Romain for starting this: a few comments:

a) As for a new definition of an event as a "physical gathering offered to an audience as a precise date and place". 

Event is a very generic concept, that we may need in multiple contexts in schema.org. The current definition 
"An event happening at a certain time and location, such as a concert, lecture, or festival. Ticketing information may be added via the 'offers' property. Repeated events may be structured as separate Event objects"
strikes a balance between a generic meaning and examples of typical, narrower definitions. I would prefer to keep the existing definition. Think of the usage of events in the Internet of Things, virtual meetings (like online sessions), historic events (e.g. being represented by a http://schema.org/MediaObject), etc.

b) I am not convinced that a straightforward subtyping of Event (if I understand your proposal correctly) is scalable. First of all, as with product types, it is very difficult to come up with a finite list of event types. Second, event type by be characterized along multiple different dimensions (e.g. age of target audience - "Events for children", or type of participation (active vs. passive - exhibition vs. workshop).

Why not simply use http://schema.org/category (by adding Event to is domain) and  http://schema.org/additionalType in combination with external type systems?

c) What about events that combine sub-events, span multiple days (or even consecutive weekends), or multiple locations?

Martin

**Comment by vholland** [comment]
@romainlange to @mfhepp's point, do you have examples for these categories?

I am trying to imagine something like SXSW or Burning Man? These don't necessarily fall into a single type.

**Comment by rvguha** [comment]
Event types is certainly on our list of things to do.

For better or worse, there are a *lot* of types of events, from Farmers
markets to birthday parties. I am not sure if the right thing to do is to
make these all classes. We might be able to get away with just making them
enumerations.

guha

**Comment by thadguidry** [comment]
@scor Isn't there an option for Drual and OpenPublic http://cgit.drupalcode.org/openpublic/log/?qt=grep&q=event
to use an enumeration list for Event Type already ?  I thought there was...do you recall how Drupal handles this or has options for site admins ?

**Comment by mfhepp** [comment]
Why not simply add schema:Event to the domain of schema:category? Then, people can use strings to indicate the type of event. With additionalProperty and www.productontology.org or any other external vocab, one could, in parallel or alternatively, indicate the exact type of event, too.

IMO, just defining a type for the enumeration values is not much better than just using a category string.

As for subclasses: I would only define subclasses of Event if there is a need for additional properties for that type.

**Comment by rvguha** [comment]
It would be good to have a canonical way for referring to different kinds
of events. Strings are always an option, but never the best option.

I do believe there is a very substantial difference between an enumeration
of allowed values and arbitrary strings.

guha

**Comment by romainlange** [comment]
Hi, thanks for your feedbacks.

Here are complementary elements:


### There are few types of events

There is great diversity of events and it is even the goal of many organizers to make new events: didn't [Red Bull make a guy jump form space?](https://www.youtube.com/watch?v=E-w2_nTnunc) 
In this case, EventType=Sport. It is just extreme skyDiving.

IMHO, it is possible to make a short list of types, then people jsut describe them.

Farmers market -> eventType=street market
Birthday parties -> eventType=Parties


### Events are physical gathering

It is now easy to attend events remotely:
- sport games can be watched in bars
- [Apple Keynote are being screened in Rotterdam](https://www.facebook.com/events/190621571114979/)
- in webinars, 100% of the audience is remote.

Each event has a physical location. For the webinar, it is the office of the main organizer.
Part or all of the audience can be remote.
It is not mandatory to include a location to be schema-compatible though.


### Schema already describes dimensions of an event

@mfhepp: schema already includes [various dimensions of the events](http://schema.org/Event) such as typical age rank of the targeted audience. I am satisfied with it.


### Festivals are superEvents

Schema includes the notion of subEvent and superEvent. SXSW and BurningMan are festivals, i.e. group of events, which can be of different types: concerts, lectures, conferences, political meetings and so on.
I added this type to the gDoc. The eventType _festival_ already exists in schema.

I think that we are close to it, do you?

**Comment by thadguidry** [comment]
Hmm,  NO @romainlange .

Your mixing things a bit.  Many of those should not be Event Types ... instead think:
Subject of an Event - http://schema.org/about

but that is currently just a problem area, where we need to add Event as a type that an about property can be used on.
We flip things over for Events and handle them more has Content or even a Creative Work, if you will... and later we can even take advantage of hierarchies from http://schema.org/category if we want.

Isn't that a much better approach than creating tons of Event Types them already seem never ending ?

**Comment by mfhepp** [comment]
@rvguha I have not problem with adding a schema:eventType property and schema:EventCategory class with pre-defined instances of EventCategory. One caveat of this is that we already use a subclass-approach for the following ones:

BusinessEvent
ChildrensEvent
ComedyEvent
DanceEvent
DeliveryEvent
EducationEvent
Festival
FoodEvent
LiteraryEvent
MusicEvent
PublicationEvent
SaleEvent
SocialEvent
SportsEvent
TheaterEvent
UserInteraction
VisualArtsEvent

So maybe a careful extension of the list of subclasses and the use of schema:category is more consistent.

**Comment by thadguidry** [comment]
@mfhepp @rvguha I agree that a basic subclass list is ideal (it might need tweaks to it in the future as nothing is constant except time)  And then as @mfhepp suggests, we can use schema:category for refinement when needed.

All those in favor of this approach ?

@mfhepp While we're waiting for the votes to come in....can you do a quick look through and make a solid 1st pass on the basic subclasses (current or those needed) so we can address the holes ?

**Comment by rvguha** [comment]
I agree with your observation. I would like to move these from being
categories to enumerations

guha

**Comment by romainlange** [comment]
I agree with @mfhepp on this.
We should keep general types and a careful extension of the list of subclasses.

**Comment by danbri** [comment]
see also #445

**Comment by romainlange** [comment]
Hi, 

we are currently working a typology of events on our platform, OpenAgenda.com.
We are also in touch with cities and media in order to get their feedbacks.

https://docs.google.com/document/d/1ITxdeLg62T4LBYc1KXDxkZ3YvjyfzSJzvY8o4AP5luw/edit?usp=sharing
This document is opened to comments and we can have further discussion here.

Thanks for your feedback!

**Comment by danbri** [comment]
Belated +1 to "I do believe there is a very substantial difference between an enumeration
of allowed values and arbitrary strings."

**Comment by vholland** [comment]
I belatedly added comments in the document. I'm happy to discuss here if that is easier.

**Comment by twamarc** [comment]
Do we include here “LifeEvent”? I am thinking about Birth, Death, maybe also BirthDay / Anniversary.  Those events are always quoted in Agenda, but also (online) newspapers, etc.

**Comment by arkhi** [comment]
> Do we include here “LifeEvent”? I am thinking about Birth, Death, maybe also BirthDay / Anniversary.

I was looking into how to identify birth in a list of events (a résumé). _LifeEvent_ could be a great addition to have a bidirectional relationship between a Person and an Event.
